libmachine/cert/cert.go

```diff
@@ -259,15 +262,28 @@ func (xcg *X509CertGenerator) ValidateCertificate(addr string, authOptions *auth
 		return false, err
 	}
 
-	dialer := &net.Dialer{
-		Timeout: time.Second * 20,
+	transport := http.Transport{
+		TLSClientConfig: tlsConfig,
 	}
 
-	_, err = tls.DialWithDialer(dialer, "tcp", addr, tlsConfig)
+	envVar := util.GetProxyURL()
+	if envVar != "" {
+		log.Debugf("proxy address is used: %s", envVar)
+		// TODO: add support for proxy server using HTTPS protocol
```

Comment:
I am not a fan of committing TODOs to the code base, we should make a separate issue in gh and call it out in the pr.

Reply:
I agree.  
In this case, because I am seeing a real need for supporting the HTTPS protocol, I will just remove the comment.